src/ape_ethereum/ecosystem.py

```diff
@@ -159,22 +160,24 @@ def decode(cls, data: dict) -> ReceiptAPI:
         )
 
 
+class BlockGasFee(BlockGasAPI):
+    @property
+    def base_fee(self) -> Optional[int]:
+        return self.raw_data.get("baseFeePerGas")
+
+    @property
+    def total_gas_used(self) -> int:
+        return self.raw_data.get("gasUsed") or 0
+
+
 class Block(BlockAPI):
     @classmethod
     def decode(cls, data: Dict) -> "BlockAPI":
-        gas_fee_data = {
-            "base_fee_per_gas": data.get("baseFeePerGas"),
-            "gas_limit": data.get("gasLimit"),
-            "gas_used": data.get("gasUsed"),
-        }
-        consensus_data = {
-            "miner": AddressType(data["miner"]) if "miner" in data else None,
-            "difficulty": data.get("difficulty"),
-            "total_difficulty": data.get("totalDifficulty"),
-        }
+        gas_fee_data = BlockGasFee(  # type: ignore
+            raw_data={k: v for k, v in data.items() if k in ["baseFeePerGas", "gasUsed"]}
```

Comment:
even for EIP1559 chains, I think pre-EIP1559 fork blocks need to handle scenarios where this data might not exist :thinking:

Reply:
What needs to happen when it does not exist?

Reply:
I think they make the blocks pre-EIP1559 compatible in some way, but I'm not sure how to handle it or if the way they are handling it makes sense for our use case. Can probably see this in web3py where they made fixes for EIP1559

Reply:
As far as I can tell, the `baseFeePerGas` just is not present on pre-London blocks. I think just returning `None` for all gas-related block properties is the safest way. `ProviderAPIs` that need to access this data will know whether it should be there or not, or handle it however if it is not.

Maybe `gas_used` does not need to be optional though. Are there any props we can always count on being part of the `BlockGasAPI` ?

Reply:
`None` seems like the path forward. `gas_used` should always be there
other things like average and total too